api/types/authentication.go

```diff
@@ -170,6 +188,16 @@ func (c *AuthPreferenceV2) SetResourceID(id int64) {
 	c.Metadata.ID = id
 }
 
+// Origin returns the origin value of the resource.
+func (c *AuthPreferenceV2) Origin() string {
```

Comment:
Nit: `GetOrigin()`? :)

Reply:
Aren't we following [official Go guidelines](https://golang.org/doc/effective_go#Getters) as much as possible/reasonable with new code?